tree/ntuple/v7/inc/ROOT/RNTupleMerger.hxx

```diff
@@ -34,10 +46,67 @@ class RFieldMerger {
 private:
    /// The merged field descriptor
    RFieldDescriptor fMergedField = RFieldDescriptor();
+
 public:
    static RResult<RFieldMerger> Merge(const RFieldDescriptor &lhs, const RFieldDescriptor &rhs);
 };
 
+// clang-format off
+/**
+ * \class ROOT::Experimental::RNTupleMerger
+ * \ingroup NTuple
+ * \brief Given a set of RPageSources merge them into an RPageSink
+ */
+// clang-format on
+class RNTupleMerger {
+
+public:
+   /// Merge a given set of sources into the destination
+   void Merge(std::vector<std::unique_ptr<RPageSource>> &sources, std::unique_ptr<RPageSink> &destination);
```

Comment:
That's a good point. I can certainly move to `std::span` and make it clear that we don't intend to get the ownership. Since we're discussing `std::span`, I just wanted to ask if I can assume we're `C++20` complaint in the master, now. Is that the case?

Reply:
I reworked this method's signature. Please let me know how the new version looks.